src/main/java/ch/uzh/ifi/hase/soprafs22/helper/Passwords.java

```diff
@@ -0,0 +1,80 @@
+package ch.uzh.ifi.hase.soprafs22.helper;
+
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import javax.validation.constraints.NotNull;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Arrays;
+import java.util.Random;
+
+// Code borrowed from https://stackoverflow.com/a/18143616/17532411 by @https://stackoverflow.com/users/829571/assylias
```

Comment:
i would not call it borrowed but something like developed on. or template by

Reply:
okay